clouddriver-elasticsearch-aws/src/main/kotlin/com/netflix/spinnaker/clouddriver/elasticsearch/aws/ElasticSearchAmazonServerGroupCachingAgent.kt

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.elasticsearch.aws
+
+
+import com.amazonaws.services.autoscaling.AmazonAutoScaling
+import com.amazonaws.services.autoscaling.model.AutoScalingGroup
+import com.amazonaws.services.autoscaling.model.DescribeAutoScalingGroupsRequest
+import com.amazonaws.services.autoscaling.model.DescribeLaunchConfigurationsRequest
+import com.amazonaws.services.autoscaling.model.LaunchConfiguration
+import com.amazonaws.services.ec2.AmazonEC2
+import com.amazonaws.services.ec2.model.DescribeInstancesRequest
+import com.amazonaws.services.ec2.model.Instance
+import com.google.common.collect.Lists
+import com.netflix.frigga.Names
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.cats.agent.RunnableAgent
+import com.netflix.spinnaker.clouddriver.aws.provider.AwsProvider
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider
+import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials
+import com.netflix.spinnaker.clouddriver.cache.CustomScheduledAgent
+import com.netflix.spinnaker.clouddriver.elasticsearch.ElasticSearchClient
+import com.netflix.spinnaker.clouddriver.elasticsearch.model.*
+import com.netflix.spinnaker.kork.core.RetrySupport
+import org.slf4j.LoggerFactory
+
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicInteger
+
+class ElasticSearchAmazonServerGroupCachingAgent(
+  val retrySupport: RetrySupport,
+  val registry: Registry,
+  val amazonClientProvider: AmazonClientProvider,
+  val accounts: Collection<NetflixAmazonCredentials>,
+  val elasticSearchClient: ElasticSearchClient
+) : RunnableAgent, CustomScheduledAgent {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(ElasticSearchAmazonServerGroupCachingAgent::class.java)
+
+    private val DEFAULT_POLL_INTERVAL_MILLIS = TimeUnit.SECONDS.toMillis(180)
+    private val DEFAULT_TIMEOUT_MILLIS = TimeUnit.MINUTES.toMillis(5)
+    private val PREFIX = "aws_server_groups"
```

Comment:
I'm not sure it's really worth putting things like this in a companion unless you need to access them statically from Java TBH.